pygbm/splitting.py

```diff
@@ -362,26 +368,29 @@ def _find_histogram_split_subtraction(context, feature_idx,
         sibling_histograms[feature_idx])
 
     return _find_best_bin_to_split_helper(
-        context, feature_idx, histogram, sum_gradients, sum_hessians)
+        context, feature_idx, histogram, sum_gradients, sum_hessians,
+        n_samples)
 
 
-@njit(locals={'gradient_left': float32, 'hessian_left': float32},
+@njit(locals={'gradient_left': float32, 'hessian_left': float32,
+              'n_samples_left': uint32},
       fastmath=True)
 def _find_best_bin_to_split_helper(context, feature_idx, histogram,
-                                   sum_gradients, sum_hessians):
+                                   sum_gradients, sum_hessians, n_samples):
```

Comment:
I had not realized but it's probably stupid to pass n_samples as an argument here. The information is already passed as part of the context.

Reply:
I don't think it's in the context, I thought about it but all the arrays are of size `n_samples` which is the size of the whole training set

Reply:
Yes, this is what I meant.

Reply:
You are right.